### PR TITLE
388 timeseries init

### LINKED ISSUE #388: TimeSeries __init__ should understand xarray structures
As the title says, a `TimeSeries` should be constructible from a xarray structure that matches the internal format. This is currently not possible but would make it much easier to convert the data from our lab.

### PR DESCRIPTION
**Pull request by vhirtham**
## Changes

`TimeSeries.__init__` accepts `xarray.DataArray` as `data` parameter.

## Related Issues

Closes #388 

## Checks

- [x] updated CHANGELOG.md
- [x] updated tests
- [x] updated doc/
- [x] ~~update example/tutorial notebooks~~

### CODE REVIEW

File: weldx/core.py [hunk 1]
```diff
@@ -346,9 +346,29 @@ def __repr__(self):
...
+            ut.to_pandas_time_index(data_array.coords.get("time").data)
+        except TypeError:
+            raise TypeError(
+                "The time values of the 'DataArray' must be convertible to a "
+                "'pandas.TimedeltaIndex'."
+            )
+
+        if not isinstance(data_array.data, pint.Quantity):
+            raise TypeError("The data of the 'DataArray' must be a 'pint.Quantity'.")
+
```
Comment (CagtayFabry) [inline weldx/core.py:368]: I think this should be a place to use `util.xr_check_coords`

File: weldx/core.py [hunk 2]
```diff
@@ -346,9 +346,29 @@ def __repr__(self):
             )
         return representation + f"Units:\n\t{self.units}\n"
 
+    @staticmethod
+    def _check_data_array(data_array: xr.DataArray):
+        """Raise an exception if the 'DataArray' can't be used as 'self._data'."""
+        if "time" != data_array.dims[0]:
+            raise ValueError("The first dimension of the DataArray must be 'time'.")
```
Comment (CagtayFabry) [inline weldx/core.py:353]: not sure if this is what you meant here but does the first dimension really have to be the time?
In that case we should just transpose the xarray correctly after the checks

File: weldx/core.py (hunk 1 above)
Comment (vhirtham) [reply weldx/core.py:368]: Not sure about this. We have to create a reference `DataArray` just to replace 3 (the time position case will be removed) simple checks. Feels to be too much overhead to me. Additionally, the error messages are probably less clear to the user.

File: weldx/core.py (hunk 2 above)
Comment (vhirtham) [reply weldx/core.py:353]: That's the internal convention for the `TimeSeries`. But you are right, if that convention is not met, we can simply transpose the dimensions. Will change it.

File: weldx/core.py (hunk 1 above)
Comment (CagtayFabry) [reply weldx/core.py]: why would we have to create another `DataArray`?
```python
xr_check_coords(data_array, {time={"dtype": ["timedelta64[ns]"]})
```

I prefer this for consistency over individual implementations just for different error messages (maybe throw one meaningful error when `xr_check_coords` fails, that should be enough)
If the error messages turn out not to be helpful enough we can reassess later

Comment (vhirtham) [reply weldx/core.py]: > why would we have to create another `DataArray`?

I somehow misread the docstring 🤦 My bad...

File: weldx/core.py [hunk 3]
```diff
@@ -357,24 +373,32 @@ def _initialize_discrete(
...
-            raise ValueError(
-                '"time" must be a time quantity or a "pandas.TimedeltaIndex".'
-            )
+        if isinstance(data, xr.DataArray):
+            self._check_data_array(data)
+            if data.dims[0] != "time":
+                data = data.transpose(
+                    *ut.swap_list_items(data.dims, data.dims[0], "time")
+                )
+            self._data = data
```
Comment (CagtayFabry) [inline weldx/core.py:382]: can't look up the code right now but there is a syntax that should work like:
~~~python
self. _data = data.transpose(..., "time") 
~~~
(check some LCS code)

**Review by CagtayFabry — CHANGES_REQUESTED** [review]

File: weldx/core.py (hunk 3 above)
Comment (vhirtham) [reply weldx/core.py:380]: works... good to know

Comment (CagtayFabry) [reply weldx/core.py:380]: I think the if check is not necessary with this

File: weldx/core.py [hunk 4]
```diff
@@ -357,24 +373,30 @@ def _initialize_discrete(
...
-        if isinstance(time, pint.Quantity):
-            time = ut.to_pandas_time_index(time)
-        if not isinstance(time, pd.TimedeltaIndex):
-            raise ValueError(
-                '"time" must be a time quantity or a "pandas.TimedeltaIndex".'
-            )
+        if isinstance(data, xr.DataArray):
+            self._check_data_array(data)
+            if data.dims[0] != "time":
+                data = data.transpose("time", ...)
```
Comment (vhirtham) [inline weldx/core.py:379]:
```suggestion
            data = data.transpose("time", ...)
```

**Review by CagtayFabry — APPROVED** [review]

**Pull request merged**